File: 2021/01/01.py
def preprocessing(input):
    return [int(item) for item in input.splitlines()]

def solver(measurements): 
    sum_one   = (measurements[-1] > measurements[-2]) + (measurements[-2] > measurements[-3])
    sum_three = 0
    for i in range(len(measurements) - 3):
        sum_one   += measurements[i + 1] > measurements[i]
        sum_three += measurements[i + 3] > measurements[i]
    yield sum_one
    yield sum_three

File: 2021/01/01.rs
aoc::main!(); 

fn preprocessing(input: &str) -> Vec<u16> {
    input
    .lines()
    .map(|measurement| measurement.parse().unwrap())
    .collect_vec()
}

fn part_1(measurements: Vec<u16>) -> usize {
    solver(measurements, 1)
}

fn part_2(measurements: Vec<u16>) -> usize {
    solver(measurements, 3)
}

fn solver(measurements: Vec<u16>, delta: usize) -> usize {
    measurements
        .iter()
        .zip(measurements.iter().skip(delta))
        .filter(|(a, b)| a < b)
        .count()
}